File: DiscordBot/src/main/java/com/example/demo/Listeners/implementations/RateListenerImpl.java
package com.example.demo.Listeners.implementations;

import com.example.demo.Listeners.RateListener;
import com.example.demo.services.MessagingService;
import org.javacord.api.entity.message.MessageBuilder;
import org.javacord.api.entity.message.embed.EmbedBuilder;
import org.javacord.api.event.message.MessageCreateEvent;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

import java.awt.*;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

@Component
public class RateListenerImpl implements RateListener {
    private final static Pattern pattern = Pattern.compile(";;rate (\\w+)");

    @Autowired
    private MessagingService messagingService;


    @Override
    public void onMessageCreate(MessageCreateEvent event) {
        if (event.getMessageContent().startsWith(";;rate")) {
            Matcher matcher = pattern.matcher(event.getMessageContent());
            if(matcher.matches()) {
                //do the matching thing
                int rating = (int) Math.floor(Math.random() * 100) + 1;

                messagingService.sendMessage(event.getMessageAuthor(),
                        "Rate calculator",
                        event.getMessageAuthor().getDisplayName()+ " is " + rating + "% " + matcher.group(1),
                        "Rate again using the same command?",
                        "https://www.nme.com/wp-content/uploads/2021/07/assassinscreed-infinity-696x442.jpg",
                        event.getChannel());


                event.getChannel().sendMessage(
                  event.getMessageAuthor().getDisplayName()
                );
            } else {
                // send the user helpful statement
                messagingService.sendMessage(event.getMessageAuthor(),
                        "Rate calculator",
                        "are you trying to use the ;;rate command? please use the correct syntax ';;rate [word]'",
                        "Rate again using the same command?",
                        null,
                        event.getChannel(), true);
            }
        }
    }
}

File: KotlinBot/src/main/kotlin/com/example/KotlinBot/Listener/Implementation/RateListenerImpl.kt
package com.example.KotlinBot.Listener.Implementation

import com.example.KotlinBot.Listener.RateListener
import com.example.KotlinBot.Service.MessagingService
import org.javacord.api.event.message.MessageCreateEvent
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Component
import java.util.regex.Matcher
import java.util.regex.Pattern
import kotlin.math.floor

@Component
class RateListenerImpl(
    @Autowired private val messagingService: MessagingService
) : RateListener {

    companion object {
        val pattern : Pattern = Pattern.compile(";;rate (\\w+)")
    }

    override fun onMessageCreate(event: MessageCreateEvent?) {
        if (event!!.messageContent.startsWith(";;rate")) {
            val matcher: Matcher = RateListenerImpl.pattern.matcher(event.messageContent)
            if (matcher.matches()) {
                //do the matching thing
                val rating = floor(Math.random() * 100).toInt() + 1
                messagingService.sendMessage(
                    event.messageAuthor,
                    "Rate calculator",
                    event.messageAuthor.displayName + " is " + rating + "% " + matcher.group(1),
                    "Rate again using the same command?",
                    "https://www.nme.com/wp-content/uploads/2021/07/assassinscreed-infinity-696x442.jpg",
                    event.channel
                )
                event.channel.sendMessage(
                    event.messageAuthor.displayName
                )
            } else {
                // send the user helpful statement
                messagingService.sendMessage(
                    event.messageAuthor,
                    "Rate calculator",
                    "are you trying to use the ;;rate command? please use the correct syntax ';;rate [word]'",
                    "Rate again using the same command?",
                    null,
                    event.channel, true
                )
            }
        }
    }
}